src/WorkflowManager/Storage/Services/DicomService.cs

```diff
@@ -113,20 +143,26 @@ public async Task<PatientDetails> GetPayloadPatientDetailsAsync(string payloadId
                     var stream = await _storageService.GetObjectAsync(bucketId, filePath);
                     var jsonStr = Encoding.UTF8.GetString(((MemoryStream)stream).ToArray());
 
-                    var dict = new Dictionary<string, DicomValue>(StringComparer.OrdinalIgnoreCase);
-                    JsonConvert.PopulateObject(jsonStr, dict);
+                    var dictCurrent = new Dictionary<string, DicomValue>(StringComparer.OrdinalIgnoreCase);
+                    JsonConvert.PopulateObject(jsonStr, dictCurrent);
 
-                    var value = GetValue(dict, keyId);
 
-                    if (!string.IsNullOrWhiteSpace(value))
+                    // merge the two dictionaries
+                    foreach (var (key, value) in dictCurrent)
                     {
-                        return value;
+                        if (dict.ContainsKey(key))
```

Comment:
Could we not re-write this to

```
if (!dict.ContainsKey(key)) {
   dict.Add(key, value);
}
```

And then no need for `continue`?

Reply:
changed to tryAdd now